src/Integration.Vsix/SonarLintTagger/TaggerProvider.cs

```diff
@@ -201,13 +200,7 @@ public void RemoveIssueTracker(IIssueTracker issueTracker)
             lock (issueTrackers)
             {
                 issueTrackers.Remove(issueTracker);
-                sonarErrorDataSource.RemoveFactory(issueTracker.Factory);
             }
         }
-
-        public void RefreshErrorList()
-        {
-            sonarErrorDataSource.RefreshErrorList();
-        }
```

Comment:
With these changes the tagger provider and tracker are now slightly less tightly coupled than they were before.

Reply:
With this change, we can almost get rid of having TaggerProvider inside TextBufferIssueTracker. I see that the remaining usage is `RequestAnalysis`, which is really bad: TaggerProvider.OnAnalysisRequested calls TextBufferIssueTracker.RequestAnalysis, which in return calls TaggerProvider.RequestAnalysis. And this is because TextBufferIssueTracker is also IIssueConsumer.

I think we can simply have TaggerProvider.OnAnalysisRequested call TaggerProvider.RequestAnalysis and pass `filteredIssueTrackers` as `IIssueConsumer`.

And then TextBufferIssueTracker won't need to know about TaggerProvider 🎉